codesign/inputparse.go

```diff
@@ -92,3 +105,52 @@ func validateCertificates(certURLList string, certPassphraseList string) ([]stri
 func splitAndClean(list string, sep string, omitEmpty bool) (items []string) {
 	return sliceutil.CleanWhitespace(strings.Split(list, sep), omitEmpty)
 }
+
+func parseFallbackProvisioningProfiles(profilesList string) ([]string, error) {
+	profiles := splitAndClean(profilesList, "\n", true)
+	if len(profiles) == 1 {
+		profiles = splitAndClean(profiles[0], "|", true)
+	}
+
+	var validProfiles []string
+	for _, profile := range profiles {
+		profileURL, err := url.Parse(profile)
+		if err != nil {
+			return nil, fmt.Errorf("invalid provisioning profile URL (%s): %w", profile, err)
+		}
+
+		if profileURL.Scheme == "file" {
+			validProfiles = append(validProfiles, profile)
+
+			continue
+		}
+
+		profilesDir := profile
+		exists, err := pathutil.IsDirExists(profilesDir)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if provisioning profile path (%s) exists: %w", profile, err)
+		} else if !exists {
+			validProfiles = append(validProfiles, profile)
+
+			continue
+		}
+
+		dirEntries, err := os.ReadDir(profilesDir)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list profile directory: %w", err)
+		}
+
+		for _, dirEntry := range dirEntries {
+			if dirEntry.Type().IsDir() || !dirEntry.Type().IsRegular() {
+				continue
+			}
+
+			if strings.HasSuffix(dirEntry.Name(), codesignasset.ProfileIOSExtension) {
+				profileURL := fmt.Sprintf("file://%s", filepath.Join(profilesDir, dirEntry.Name()))
+				validProfiles = append(validProfiles, profileURL)
+			}
+		}
```

Comment:
this could be moved to a separate function, to make this function cleaner.

Reply:
Agree, updated.